packages/transaction-controller/src/TransactionController.ts

```diff
@@ -963,6 +968,63 @@ export class TransactionController extends BaseController<
     }
   }
 
+  updateTransactionGasFees(
+    transactionID: string,
+    {
+      gas,
+      gasLimit,
+      gasPrice,
+      maxPriorityFeePerGas,
+      maxFeePerGas,
+      estimateUsed,
+      estimateSuggested,
+      defaultGasEstimates,
+      originalGasEstimate,
+      userEditedGasLimit,
+      userFeeLevel,
+    }: TransactionGasValues & TransactionMetaGasValues,
```

Comment:
I appreciate the intent here is to encapsulate all the gas fields in types so we can group them, though I fear it's introducing more complexity than it's saving.

This is because it causes our primary `TransactionMeta` and `Transaction` types to split and be the product of multiple unions, so it's no longer possible to simply look at all the properties in a list which I think is valuable for such big and fundamental types.

It would also make this method a little easier to maintain as its type can be defined inline, plus there would only be one usage of these types at the moment anyway.

Reply:
Moved them back to inline types https://github.com/MetaMask/core/pull/1674/commits/171ac9f32f15a5730713fd916b612c94d0ec6559